crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -71,7 +71,7 @@ impl TradeEstimator {
                 finder,
                 verifier: None,
             }),
-            sharing: Default::default(),
+            sharing: RequestSharing::labelled("TradeEstimator".into()),
```

Comment:
The way the `TradeFinder` price estimator works is that it uses an inner `TradeFinding` implementation (so the inner `trade_finding/{oneinch,paraswap,etc.}.rs`) to find a `Trade` and turns that into a price estimate (its like an adapter between a DEX aggregator-like API and the `PriceEstimating` interface we use internally).

This means that many price estimators will share this "TradeEstimator" label and we will lose information depending if queries are reused at the DEX API level or at the price estimation level.

Can we add a:

```rust
impl TradeEstimator {
    pub fn with_sharing_label(mut self, label: String) -> Self {
        self.sharing = RequestSharing::labelled(label);
        self
    }
}
```

And then change `price_estimation/{zeroex,oneinch,etc.}.rs` to add a different label per price estimator?